conftest.py

```diff
@@ -44,6 +44,10 @@ def pytest_collection(session):
 
     This function runs before collecting tests.
     """
+    # setting path constants to None on utils module is a dirty way to ensure
+    # no test will write on top of user config file. This is a last resort, as
+    # mock_path_constants fixture shall be doing the job of actually patching
+    # those to a usable value in tests
```

Comment:
Thank you for explaining this mystery function call!

Reply:
almost 2 years ago me was FURIOUS after realising that running tests modified my local configuration 😅